### ISSUE TITLE
Error: Running LUA method 'update' - mode3.lua:24: attempt to perform arithmetic on a nil value

### ISSUE THREAD

**Issue by vjacksonr123** [Courseplay/courseplay#4125]
At the End of a course I am having a lock up of course play happening when it just goes into a loop and starts writing lots of error to the log file. 

This happens sometimes when I click the vehicle icon above the mini map, or tab to the vehicle. 

When this happens, windows defender also goes crazy. I close farm sim and everything goes back to normal. 

I have had this now multiple times, one with the Big Brute from the mod store, and many times with the JD 9RT from mod store.  These are the ones that stick out, the one attached is from the brute.  

Here is an excerpt from the error log when it occurred on me last night:  
[log.txt](https://github.com/Courseplay/courseplay/files/3521757/log.txt)

2019-08-20 01:53 CourseGeneratorScreen:onCreate()
2019-08-20 02:00 ## Courseplay: aiTrafficCollisionTrigger missing. Traffic collision prevention will not work! vehicle 425/100 (#5)
2019-08-20 02:00 ## Courseplay: aiTrafficCollisionTrigger missing. Traffic collision prevention will not work! vehicle 425/100 (#5)
2019-08-20 02:10 Warning (script): Unknown entity id 1510136 in method 'getParent'.
2019-08-20 02:10 LUA call stack:
2019-08-20 02:10   dataS/scripts/vehicles/Vehicle.lua (1703) : getParent
2019-08-20 02:10   dataS/scripts/vehicles/specializations/Suspensions.lua (259) : getParentComponent
2019-08-20 02:10   dataS/scripts/vehicles/specializations/Suspensions.lua (276) : setSuspensionNodeCharacter
2019-08-20 02:10   dataS/scripts/vehicles/SpecializationUtil.lua (30)
2019-08-20 02:10   dataS/scripts/vehicles/specializations/Enterable.lua (621) : raiseEvent
2019-08-20 02:10   dataS/scripts/BaseMission.lua (1881) : enterVehicle
2019-08-20 02:10   dataS/scripts/FSBaseMission.lua (3221) : onEnterVehicle
2019-08-20 02:10   dataS/scripts/vehicles/VehicleEnterResponseEvent.lua (66) : onEnterVehicle
2019-08-20 02:10   dataS/scripts/network/Connection.lua (77) : run
2019-08-20 02:10   dataS/scripts/vehicles/VehicleEnterRequestEvent.lua (68) : sendEvent
2019-08-20 02:10   dataS/scripts/network/Connection.lua (77) : run
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/settings.lua (1397) : sendEvent
2019-08-20 02:10   setfenv(1,Courseplay_6_01_00283); courseplay:goToVehicle(...) (1) : goToVehicle
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/input.lua (189)
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/button.lua (232) : executeFunction
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/CpManager.lua (364) : handleMouseClick
2019-08-20 02:10   dataS/scripts/BaseMission.lua (1418) : mouseEvent
2019-08-20 02:10   dataS/scripts/FSBaseMission.lua (1921) : mouseEvent
2019-08-20 02:10   dataS/scripts/main.lua (1754) : mouseEvent
2019-08-20 02:10 Warning (script): 'getParent': Argument 1 has wrong type. Expected: Int. Actual: Nil 
2019-08-20 02:10   ..\..\..\src\base\scripting\lua\LuaScriptSystem.cpp (882): expectedType == Value::VoidType || expectedType == Value::ClassType
2019-08-20 02:10 LUA call stack:
2019-08-20 02:10   dataS/scripts/vehicles/Vehicle.lua (1703) : getParent
2019-08-20 02:10   dataS/scripts/vehicles/specializations/Suspensions.lua (259) : getParentComponent
2019-08-20 02:10   dataS/scripts/vehicles/specializations/Suspensions.lua (276) : setSuspensionNodeCharacter
2019-08-20 02:10   dataS/scripts/vehicles/SpecializationUtil.lua (30)
2019-08-20 02:10   dataS/scripts/vehicles/specializations/Enterable.lua (621) : raiseEvent
2019-08-20 02:10   dataS/scripts/BaseMission.lua (1881) : enterVehicle
2019-08-20 02:10   dataS/scripts/FSBaseMission.lua (3221) : onEnterVehicle
2019-08-20 02:10   dataS/scripts/vehicles/VehicleEnterResponseEvent.lua (66) : onEnterVehicle
2019-08-20 02:10   dataS/scripts/network/Connection.lua (77) : run
2019-08-20 02:10   dataS/scripts/vehicles/VehicleEnterRequestEvent.lua (68) : sendEvent
2019-08-20 02:10   dataS/scripts/network/Connection.lua (77) : run
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/settings.lua (1397) : sendEvent
2019-08-20 02:10   setfenv(1,Courseplay_6_01_00283); courseplay:goToVehicle(...) (1) : goToVehicle
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/input.lua (189)
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/button.lua (232) : executeFunction
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/CpManager.lua (364) : handleMouseClick
2019-08-20 02:10   dataS/scripts/BaseMission.lua (1418) : mouseEvent
2019-08-20 02:10   dataS/scripts/FSBaseMission.lua (1921) : mouseEvent
2019-08-20 02:10   dataS/scripts/main.lua (1754) : mouseEvent
2019-08-20 02:10 Warning (script): 'getParent': Argument 1 has wrong type. Expected: Int. Actual: Nil 
2019-08-20 02:10   ..\..\..\src\base\scripting\lua\LuaScriptSystem.cpp (882): expectedType == Value::VoidType || expectedType == Value::ClassType
2019-08-20 02:10 LUA call stack:
2019-08-20 02:10   dataS/scripts/vehicles/Vehicle.lua (1703) : getParent
2019-08-20 02:10   dataS/scripts/vehicles/specializations/Suspensions.lua (259) : getParentComponent
2019-08-20 02:10   dataS/scripts/vehicles/specializations/Suspensions.lua (276) : setSuspensionNodeCharacter
2019-08-20 02:10   dataS/scripts/vehicles/SpecializationUtil.lua (30)
2019-08-20 02:10   dataS/scripts/vehicles/specializations/Enterable.lua (621) : raiseEvent
2019-08-20 02:10   dataS/scripts/BaseMission.lua (1881) : enterVehicle
2019-08-20 02:10   dataS/scripts/FSBaseMission.lua (3221) : onEnterVehicle
2019-08-20 02:10   dataS/scripts/vehicles/VehicleEnterResponseEvent.lua (66) : onEnterVehicle
2019-08-20 02:10   dataS/scripts/network/Connection.lua (77) : run
2019-08-20 02:10   dataS/scripts/vehicles/VehicleEnterRequestEvent.lua (68) : sendEvent
2019-08-20 02:10   dataS/scripts/network/Connection.lua (77) : run
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/settings.lua (1397) : sendEvent
2019-08-20 02:10   setfenv(1,Courseplay_6_01_00283); courseplay:goToVehicle(...) (1) : goToVehicle
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/input.lua (189)
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/button.lua (232) : executeFunction
2019-08-20 02:10   =F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/Courseplay_6_01_00283/CpManager.lua (364) : handleMouseClick
2019-08-20 02:10   dataS/scripts/BaseMission.lua (1418) : mouseEvent
2019-08-20 02:10   dataS/scripts/FSBaseMission.lua (1921) : mouseEvent
2019-08-20 02:10   dataS/scripts/main.lua (1754) : mouseEvent
2019-08-20 02:10 Warning (script): 'getParent': Argument 1 has wrong type. Expected: Int. Actual: Nil 
2019-08-20 02:10   ..\..\..\src\base\scripting\lua\LuaScriptSystem.cpp (882): expectedType == Value::VoidType || expectedType == Value::ClassType

Error log is attached. 

[log.txt](https://github.com/Courseplay/courseplay/files/3521758/log.txt)

**Comment by Tensuko** [comment]
That Modfiel say unzip me:
Error: Failed to open xml file 'F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/entpacken_bitte/modDesc.xml'.
Load mod: entpacken_bitte (Hash: 2b1db9e714811b2f7a884a7cca42a132)

Error: Failed to open xml file 'F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/BalersNadal/modDesc.xml'.
Load mod: BalersNadal (Hash: 86dd1b638e292ad5977830bbe3f176db)
Error: Missing descVersion attribute in mod BalersNadal

Error: Invalid mod name 'fs 19 silage mods'! Characters allowed: (_, A-Z, a-z, 0-9). The first character must not be a digit

Error: Unsupported mod description version in mod FS17_KroneBigM500wide_V7

Error: Unsupported mod description version in mod FS17_Large_BaleStorage_placeable

Error: Failed to open xml file 'F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/FS19_NHCR1090ByStevie/modDesc.xml'.
Load mod: FS19_NHCR1090ByStevie (Hash: 2d479cb40b88f8a925f7e47e06a1094c)
Error: Missing descVersion attribute in mod FS19_NHCR1090ByStevie

Error: Failed to open xml file 'F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/PETERBILT________SERVICETRUCK/modDesc.xml'.
Load mod: PETERBILT________SERVICETRUCK (Hash: 3508dedff7c1c7d27c370a697e882347)
Error: Missing descVersion attribute in mod PETERBILT________SERVICETRUCK

Error: Failed to open xml file 'F:/File Store/Users/Shivan/Documents/My Games/FarmingSimulator2019/mods/unzip_peterbilt_trucks_pack/modDesc.xml'.
Load mod: unzip_peterbilt_trucks_pack (Hash: 9ac5f13ca3bd752603f244b3cd6a6648)
Error: Missing descVersion attribute in mod unzip_peterbilt_trucks_pack

Load mod: ZZZ_GPS (Version: 3.2.1) (Hash: af972145274e23d5688b68c25d6159c6)
Error: Unsupported mod description version in mod ZZZ_GPS

Clean up your modfolder 1st, there are 100% a lot of Mods that are not working with CP.
That LUA Callstack can come from anything!

Similar to #4115

**Comment by vjacksonr123** [comment]
yea I will clean it up tonight and just run big bud and course play and see if i can reproduce again tonight.

**Comment by vjacksonr123** [comment]
I unloaded all mods except Courseplay 6.01.00283, Big_Brute and MoneyTool. 

I was able to replicate the issue on ravenport running multiple course play courses at one time, with just in game equipment running (except the big brute).  

Error in the log is a little different from when the game locked up but happened when I tabbed to a different machine that had been running a courseplay course. 

Snipit from the log: 
2019-08-20 22:34 ## Courseplay: aiTrafficCollisionTrigger missing. Traffic collision prevention will not work! vehicle 425/100 (#4)
2019-08-20 22:34 ## Courseplay: aiTrafficCollisionTrigger missing. Traffic collision prevention will not work! vehicle 425/100 (#4)
2019-08-20 22:35 Error: Running LUA method 'update'.
2019-08-20 22:35 F:/FS19 Custom Mods/CP Test/Courseplay_6_01_00283/mode3.lua:24: attempt to perform arithmetic on a nil value
2019-08-20 22:35 Error: Running LUA method 'update'.
2019-08-20 22:35 F:/FS19 Custom Mods/CP Test/Courseplay_6_01_00283/mode3.lua:24: attempt to perform arithmetic on a nil value
2019-08-20 22:35 Error: Running LUA method 'update'.
2019-08-20 22:35 F:/FS19 Custom Mods/CP Test/Courseplay_6_01_00283/mode3.lua:24: attempt to perform arithmetic on a nil value
2019-08-20 22:35 Error: Running LUA method 'update'.
2019-08-20 22:35 F:/FS19 Custom Mods/CP Test/Courseplay_6_01_00283/mode3.lua:24: attempt to perform arithmetic on a nil value
2019-08-20 22:35 Error: Running LUA method 'update'.

I have attached the full log file. 

[log.txt](https://github.com/Courseplay/courseplay/files/3523471/log.txt)
